common/logger/src/kv.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Key-Value definitions
+
+use serde::Serialize;
+use std::fmt;
+
+#[derive(Clone, Copy, Debug, Hash, Eq, PartialEq, Ord, PartialOrd, Serialize)]
+pub struct Key(&'static str);
+
+impl Key {
+    pub fn new(s: &'static str) -> Self {
+        Self(s)
+    }
+
+    pub fn as_str(&self) -> &'static str {
+        self.0
+    }
+}
+
+#[derive(Clone)]
+pub enum Value<'v> {
```

Comment:
What's the advantage of having a Value with a lifetime versus extracting out the String?

Reply:
I'm trying to have it be a little flexible for things we might want to do in the backend. This way you're not forced to do an allocation unless the backend really wants to, which is a win